data_structures/avltree.cpp

```diff
@@ -6,38 +6,52 @@
  * \warning This program is a poor implementation and does not utilize any of
  * the C++ STL features.
  */
-#include <algorithm>
-#include <iostream>
-#include <queue>
+#include <algorithm>  // for std::max
+#include <iostream>   // for std::cout
+#include <queue>      // for std::queue
 
-typedef struct node {
+using node = struct node {
     int data;
     int height;
     struct node *left;
     struct node *right;
-} node;
+};
 
-/** Create and return a new Node */
+/**
+ * @brief creates and returns a new node
+ * @param[in] data value stored in the node
+ * @return newly created node
+ */
 node *createNode(int data) {
     node *nn = new node();
     nn->data = data;
     nn->height = 0;
-    nn->left = NULL;
-    nn->right = NULL;
+    nn->left = nullptr;
+    nn->right = nullptr;
     return nn;
 }
 
-/** Returns height of tree */
+/**
+ * @param[in] root the root of the tree
+ * @return height of tree
+ */
 int height(node *root) {
-    if (root == NULL)
+    if (root == nullptr) {
         return 0;
+    }
     return 1 + std::max(height(root->left), height(root->right));
 }
 
-/** Returns difference between height of left and right subtree */
+/**
+ * @param[in] root of the tree
+ * @return difference between height of left and right subtree
+ */
 int getBalance(node *root) { return height(root->left) - height(root->right); }
```

Comment:
Should it be absolute of difference ?

Reply:
No. Actually the sign of the result is essential. Have a look [here](https://github.com/TheAlgorithms/C-Plus-Plus/blob/b89d1da2f1250c28bbfc6290fe5594de8ec2896e/data_structures/avltree.cpp#L74).